internal/util/argParser.go

```diff
@@ -13,6 +13,7 @@ func CleanInputArguments(arguments []string) ([]string, error) {
 	for _, argument := range arguments {
 		if strings.HasPrefix(argument, "--") && strings.Contains(argument, "=") {
 			param, value, _ := strings.Cut(argument, "=")
+			// TODO: this misses values consisting of only (more than one) spaces
```

Comment:
True, the whole parameter parsing is a bit of a hack. If we choose to go forward with this approach the separate cases should be refactored into smaller utility functions to keep the code legible and maintainable.
There is quite a few edge cases with the key=value version of input passing.
Multiple spaces shouldn't be an issue since this part of the code only aims to correct inputs in the form of `key=value` and `key v a l u e`. This check only aims to correct in the case of a parameter in the form of `--key= v a l u e`
On the other hand if the parameter would get passed as `--key = v a l u e` the function would fail to correct it (purposely) it should interpret it as "key" : "= v a l u e" since if the "=" is not part of the key value it's not possible to accurately determine if the "=" should be part of the value or not.

Reply:
Clarification: 
The function basically iterates over all arguments, if it sees a "--"-prefix it recognizes it as the beginning of key, when that happens it checks if the key contains a "=" if it does it removes it and adds up all of the next arguments as the corresponding values until it finds an argument with the "--"-prefix, from where the whole thing starts again.
The check your referencing is to check if the result of cut(removing the "=") as a value attached in the case of key=v a l u e.